scalameta/parsersJSFacade/src/main/scala/scala/meta/parsersjsfacade/Api.scala

```diff
@@ -0,0 +1,59 @@
+package scala.meta
+package parsersjsfacade
+
+import scala.scalajs.js
+import js.JSConverters._
+import js.annotation._
+
+import parsers._
+
+object ScalaParser {
+
+  // https://stackoverflow.com/a/36573183/846273
+  private[this] def mergeJSObjects(objs: js.Dynamic*): js.Dynamic = {
+    val result = js.Dictionary.empty[Any]
+    for (source <- objs) {
+      for ((key, value) <- source.asInstanceOf[js.Dictionary[Any]])
+        result(key) = value
+    }
+    result.asInstanceOf[js.Dynamic]
+  }
+
+  def toNode(t: Tree): js.Dynamic = {
+    val base = js.Dynamic.literal(
+      "type" -> t.productPrefix,
+      "children" -> t.children.map(toNode).toJSArray,
+      "pos" -> js.Dynamic.literal(
+        "start" -> t.pos.start.offset,
+        "end" -> t.pos.end.offset
+      )
+    )
+
+    // FIXME: this is really just for personal gratification
+    // ideally `.toMap` on Tree will take care of adding these
+    // attributes
+    val specific = t match {
+      case t: Lit => js.Dynamic.literal(
+        "value" -> t.value.toString
```

Comment:
We may want to special case float and double since those store the value as a string https://github.com/scalameta/scalameta/blob/33c47b6fc9be8fb0bce60c17e4a4933379c23870/scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala#L311-L313 and calling `.value` on them will actually loose precision in JS runtimes. This was the biggest breaking change necessary to port scalameta to Scala.js, before there was a single `class Lit(value: Any)`